### ISSUE TITLE
test harness doesn't permit specifying the end column of a diagnostic

### ISSUE THREAD

**Issue by nikomatsakis** [dada-lang/dada#84]
I left some code in the test harness to specify the columns of a diagnostic but never modified the "expected diagnostics" function to fill that field in. I expected to parse something like this

```
fn foo() {
    blah = 3
    blah += baz
    #!      ^^^ ERROR unknown identifier
}
```

and then have the system check that the highlight was as expected.

To make this work, the first thing is to modify the `expected_diagnostics` function:

https://github.com/dada-lang/dada/blob/6dcc38a60546b499d862ced9d46f8c7fc6996ac7/components/dada-lang/src/test_harness.rs#L451

so that the regular expression can match the above and then set these fields to something other than `None`:

https://github.com/dada-lang/dada/blob/6dcc38a60546b499d862ced9d46f8c7fc6996ac7/components/dada-lang/src/test_harness.rs#L472-L473

After that, the rest of the code might need modification too, not sure.

**Comment by XFFXFF** [comment]
Maybe we should fix #52 first?

**Comment by nikomatsakis** [comment]
@XFFXFF I actually did fix that...though maybe it's in my branch and not landed.
